dev-libs/tvision/tvision-2024.02.04~git.recipe

```diff
@@ -0,0 +1,103 @@
+SUMMARY="A modern port of Turbo Vision 2.0"
+DESCRIPTION="A modern port of Turbo Vision 2.0, the classical framework for text-based user interfaces. \
+Now cross-platform and with Unicode support. "
+HOMEPAGE="https://github.com/magiblot/tvision"
+COPYRIGHT="2019-2021 magiblot"
+LICENSE="TVISION
+	MIT"
+REVISION="1"
+srcGitRev="be6e64f85355c6c96b7c09174986fafb6a411c6a"
+SOURCE_URI="https://github.com/magiblot/tvision/archive/$srcGitRev.tar.gz"
+CHECKSUM_SHA256="cb9601551831861f06ec5e6bfefe9340b5ca0fea7eeb253630ece98790620ce7"
+SOURCE_DIR="tvision-$srcGitRev"
+
+PATCHES="tvision-$portVersion.patchset"
+
+ARCHITECTURES="all !x86_gcc2"
+SECONDARY_ARCHITECTURES="x86"
+
+commandSuffix=$secondaryArchSuffix
+commandBinDir=$binDir
+if [ "$targetArchitecture" = x86_gcc2 ]; then
+	commandSuffix=
+	commandBinDir=$prefix/bin
+fi
+
+PROVIDES="
+	tvision$secondaryArchSuffix = $portVersion
+	lib:libtvision$secondaryArchSuffix = $portVersion
+	"
+REQUIRES="
+	haiku$secondaryArchSuffix
+	lib:libncursesw$secondaryArchSuffix
+	"
+
+PROVIDES_devel="
+	tvision${secondaryArchSuffix}_devel = $portVersion
+	devel:libtvision$secondaryArchSuffix = $portVersion
+	"
+REQUIRES_devel="
+	tvision$secondaryArchSuffix == $portVersion base
+	"
+
+PROVIDES_tools="
+	tvision${secondaryArchSuffix}_tools = $portVersion
+	cmd:tvdemo$commandSuffix = $portVersion
+	cmd:tvedit$commandSuffix = $portVersion
+	cmd:tvhc$commandSuffix = $portVersion
+	"
+REQUIRES_tools="
+	$REQUIRES
+	tvision$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libncursesw$secondaryArchSuffix
```

Comment:
As the test are run during BUILD it needs gtest also (both arch's), you could add it as disabled and mention that in the BUILD section also.